graylog2-web-interface/src/views/logic/aggregationbuilder/visualizations/BarVisualizationConfig.js

```diff
@@ -49,10 +49,11 @@ export default class BarVisualizationConfig extends VisualizationConfig {
   }
 }
 
+type InternalBuilderState = Immutable.Map<string, any>;
```

Comment:
That would mean that every value in the state of the builder would be of type `BarVisualizationConfigType`, which is not the case. Unfortunately we cannot really type the builder state stricter, because we have different types for the values of different keys.

I have stumbled across this too and was thinking a bit about it. We could use either JS plain objects or [Immutable Records](https://immutable-js.github.io/immutable-js/docs/#/Record) for this, but it requires changing the code and is therefore out of scope for this PR.

Reply:
Sounds reasonable, thanks for the clarification.